src/api/routes/rosetta/construction.ts

```diff
@@ -371,8 +370,10 @@ export function createRosettaConstructionRouter(db: DataStore, chainId: ChainID)
     const stxAddress = options.sender_address;
 
     // Getting nonce info
-    const accountInfo = await new StacksCoreRpcClient().getAccount(stxAddress);
-    const nonce = accountInfo.nonce;
+    const nondeNonce = await new StacksCoreRpcClient().getAccountNonce(stxAddress);
+
+    const apiNonce = await db.getAddressNonces({ stxAddress: stxAddress });
+    const nonce = Math.max(nondeNonce, apiNonce.possibleNextNonce);
```

Comment:
Should this nonce fetching logic be moved into a shared function in one of the rosetta helper files?